packages/ember-cli-code-coverage/index.js

```diff
@@ -1,85 +1,23 @@
 'use strict';
 
-var path = require('path');
-var fs = require('fs-extra');
 var attachMiddleware = require('./lib/attach-middleware');
-var config = require('./lib/config');
-const walkSync = require('walk-sync');
-const VersionChecker = require('ember-cli-version-checker');
-
-function getPlugins(appOrAddon) {
-  let options = (appOrAddon.options = appOrAddon.options || {});
-  options.babel = options.babel || {};
-  return (options.babel.plugins = options.babel.plugins || []);
-}
-
-// Regular expression to extract the file extension from a path.
-const EXT_RE = /\.[^.]+$/;
 
 module.exports = {
   name: require('./package').name,
 
-  /**
-   * Look up the file path from an ember module path.
-   * @type {Object<String, String>}
-   */
-  fileLookup: null,
-
-  // Ember Methods
-  init() {
-    this._super.init.apply(this, arguments);
-    this.fileLookup = {};
-  },
-
-  included(appOrAddon) {
-    this._super.included.apply(this, arguments);
-
-    this.parentRegistry = appOrAddon.registry;
-
-    if (!this._registeredWithBabel && this._isCoverageEnabled()) {
-      let checker = new VersionChecker(this.parent).for(
-        'ember-cli-babel',
-        'npm'
-      );
+  buildBabelPlugin(opts = {}) {
+    let project = opts.project;
+    let cwd = opts.cwd || project.root;
+    let include = opts.include || '**/*';
+    let exclude = opts.exclude || ['*/mirage/**/*'];
 
-      if (checker.satisfies('>= 6.0.0')) {
-        const IstanbulPlugin = require.resolve('babel-plugin-istanbul');
-        const exclude = this._getExcludes();
-        const include = this._getIncludes();
-
-        [this.app, this._findCoveredAddon(), ...this._findInRepoAddons()]
-          .filter(Boolean)
-          .map(getPlugins)
-          .forEach((plugins) =>
-            plugins.push([IstanbulPlugin, { exclude, include }])
-          );
-      } else {
-        this.project.ui.writeWarnLine(
-          'ember-cli-code-coverage: You are using an unsupported ember-cli-babel version,' +
-            'instrumentation will not be available.'
-        );
-      }
-
-      this._registeredWithBabel = true;
-    }
-  },
-
-  contentFor(type) {
-    if (type === 'test-body-footer' && this._isCoverageEnabled()) {
-      var template = fs
-        .readFileSync(
-          path.join(__dirname, 'lib', 'templates', 'test-body-footer.html')
-        )
-        .toString();
-      return template.replace(
-        '{%ENTRIES%}',
-        JSON.stringify(
-          Object.keys(this.fileLookup).map((file) => file.replace(EXT_RE, ''))
-        )
-      );
+    if (opts.embroider === true) {
+      let { stableWorkspaceDir } = require('@embroider/compat/src/default-pipeline');
+      cwd = stableWorkspaceDir(project.root);
     }
 
-    return undefined;
+    const IstanbulPlugin = require.resolve('babel-plugin-istanbul');
+    return [IstanbulPlugin, { cwd, include, exclude }];
```

Comment:
Let's make the consumer do `plugins: [...buildBabelPlugin()]` just in case we need to begin emitting multiple plugins in the future...